### ISSUE TITLE
Normalise the flags to disable features

### ISSUE THREAD

**Issue by rossjones** [bacalhau-project/bacalhau#3662]
We currently have a few flags we can pass to `serve` to turn off individual features. 
Unfortunately, one of the flags is ever so slightly different and it might be best to make it more consistent.

```
--disable-engine       Engine types to disable
--disable-storage      Storage types to disable
--disabled-publisher   Publisher types to disable
```

**Comment by frrist** [comment]
@wdbaruni when I completed this change do you want to deprecate (hide, but don't remove) the `--disabled-publisher` and replace it with `--disable-publisher`? Or just rename it?

**Comment by wdbaruni** [comment]
as part of https://www.notion.so/expanso/Rethinking-Configuration-435fbe87419148b4bbc5119d413786eb, we will remove the CLI flags and replace them with a `Publishers.Disabled` configuration. 

I just assigned this to you so that you can mark it as closed when #4014 is delivered, but maybe better to close this issue now to avoid confusion
